asm-utils/src/data_parser.rs

```diff
@@ -14,76 +16,134 @@ pub enum DataValue {
 
 impl DataValue {
     /// Returns the size of the value in bytes.
-    pub fn size(&self) -> usize {
+    ///
+    /// The address is necessary because the size of the alignment padding
+    /// depending on what address it is defined on.
+    pub fn size(&self, from_addr: usize) -> usize {
         match self {
             DataValue::Direct(data) => data.len(),
             DataValue::Zero(length) => *length,
+            DataValue::Alignment(bytes) => alignment_size(from_addr, *bytes),
             DataValue::Reference(_) => 4,
             DataValue::Offset(..) => 4,
         }
     }
 }
 
+#[derive(Default)]
+struct DataSections {
+    sections: Vec<Vec<(Option<String>, Vec<DataValue>)>>,
+}
+
+impl DataSections {
+    fn new() -> Self {
+        Default::default()
+    }
+
+    fn curr_entry(&mut self) -> &mut Vec<DataValue> {
```

Comment:
```suggestion
    fn current_entry(&mut self) -> &mut Vec<DataValue> {
```

This wasn't applied from the previous review